src/request.rs

```diff
@@ -909,6 +909,41 @@ impl<'a> Request<'a> {
             self.inner.send(&mut stream, writer)
         }
     }
+
+    ///Sends HTTP request.The connection will only be established if the supplied certificate
+    ///matches the server's root certificate.
+    ///Supply the content of the server's root certificate as a String. There are no predefined trusted
+    ///root certificates in the tls config.
+    ///
+    ///Creates `TcpStream` (and wraps it with `TlsStream` if needed). Writes request message
+    ///to created stream. Returns response for this request. Writes response's body to `writer`.
```

Comment:
This comment is basically just a prose version of the implementation below, which is not needed. We should primarily document the interface and any pre- and post-conditions if they're special.

Reply:
Comment from original lib, I will remove it